### PR TITLE
Upgrade to Snakebids 0.8.0

### LINKED ISSUE #72: Upgrade snakebids
## The problem

Update snakebids to `0.8.x` to take advantage of new features. Should also allow us to fix the filter / wildcard bug for the `part` entity. 

There is a conflict with `labelmerge` dependencies, so will have to hold wait for that to also be upgraded to `0.8.x`

## Environment

* Operating System
* SCATTR version

### PR DESCRIPTION
**Pull request by kaitj**
## Proposed changes

This PR resolves #72, upgrading Snakebids to `0.8.0`. Also updates the majority of the expand calls to make use of the new syntax, which cleans up quite a few lines of code.

~_(Currently waiting on bug fixes (e.g. cli filtering) for snakebids 0.8.0)_~

### TODO (prior to review)
- [x] Test workflow on real data
- [x] Finish converting `expand` syntax in relevant rules

## Types of changes

What types of changes does your code introduce? Put an `x` in the boxes that apply

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other (if none of the other choices apply)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you are unsure about any of the choices, don't hesitate to ask!

- [x] Changes have been tested to ensure that fix is effective or that a feature works.
- [x] Changes pass the unit tests
- [x] Code has been run through the `poe quality` task
- [x] I have included necessary documentation or comments (as necessary)
- [x] Any dependent changes have been merged and published

## Notes
All PRs will undergo the unit testing before being reviewed. You may be requested to explain or make additional changes before the PR is accepted.

### CODE REVIEW

File: scattr/config/snakebids.yml [hunk 1]
```diff
@@ -273,13 +273,11 @@ zona_bb_subcortex:
...
 
-# NOTE: This may not be needed with labelmerge
 freesurfer:
   tsv: 'resources/freesurfer/desc-FreesurferThal_dseg.tsv'
 
 # Containers
-# NOTE: Update from latest to use specific tags
 singularity:
-  scattr: "docker://khanlab/scattr:v0.2.1"
+  scattr: /project/6050199/tkai/singularity/scattr_v0.2.1.sif # "docker://khanlab/scattr:v0.2.1"
```
Comment (tkkuehn) [inline scattr/config/snakebids.yml:281]: This should probably be the docker URL in the repo

File: scattr/workflow/Snakefile [hunk 2]
```diff
@@ -69,12 +68,10 @@ include: "rules/qc.smk"
 
 rule all:
     input:
-        tck_files=expand(
-            rules.filtered_tck2connectome.output.sl_assignment,
-            zip,
-            **subj_zip_list
+        tck_files=inputs["T1w"].expand(
+            rules.filtered_tck2connectome.output.sl_assignment
```
Comment (tkkuehn) [inline scattr/workflow/Snakefile:72]: This going to expand over all the wildcards in `"T1w"`, not just subject and session like it did before. Is that intentional?

File: scattr/workflow/rules/qc.smk [hunk 3]
```diff
@@ -71,7 +71,7 @@ rule registration_qc:
         fixed_nii=lambda wildcards: expand(
             inputs["T1w"].path,
             zip,
-            **filter_list(inputs["T1w"].input_zip_lists, wildcards)
+            **filter_list(inputs["T1w"].zip_lists, wildcards)
```
Comment (tkkuehn) [inline scattr/workflow/rules/qc.smk:74]: I think we could make this `fixed_nii=lambda wildcards: inputs["T1w"].filter(**wildcards).expand()`

File: scattr/workflow/rules/zona_bb_subcortex.smk [hunk 4]
```diff
@@ -74,7 +74,7 @@ rule reg2native:
         target=lambda wildcards: expand(
             inputs["T1w"].path,
             zip,
-            **filter_list(inputs["T1w"].input_zip_lists, wildcards)
+            **filter_list(inputs["T1w"].zip_lists, wildcards)
```
Comment (tkkuehn) [inline scattr/workflow/rules/zona_bb_subcortex.smk:77]: I think we could make this `target=lambda wildcards: inputs["T1w"].filter(**wildcards).expand()`

**Review by tkkuehn — CHANGES_REQUESTED** [review]
A couple of suggestions to make things even terser and a couple of questions, but this otherwise looks good!

File: scattr/config/snakebids.yml (hunk 1 above)
Comment (kaitj) [reply scattr/config/snakebids.yml:281]: Ahh that is my mistake. Didn’t mean to push this change in - was meant for local testing.

File: scattr/workflow/rules/qc.smk [hunk 5]
```diff
@@ -29,7 +29,7 @@ rule segment_qc:
         t1w_image=lambda wildcards: expand(
             inputs["T1w"].path,
             zip,
-            **filter_list(inputs["T1w"].input_zip_lists, wildcards)
+            **filter_list(inputs["T1w"].zip_lists, wildcards)
```
[the remark this answers is not in the document]
Comment (kaitj) [reply scattr/workflow/rules/qc.smk:32]: Ahh I was wondering about this. I’ll play around with this.

File: scattr/workflow/Snakefile (hunk 2 above)
Comment (kaitj) [reply scattr/workflow/Snakefile:72]: Sort of - would it expand over those additional wildcards if they aren’t present in the file though?

Comment (tkkuehn) [reply scattr/workflow/Snakefile:72]: I _think_ `expand` errors out if you give it wildcards not in the path it's trying to expand, but am not sure

Comment (kaitj) [reply scattr/workflow/Snakefile:72]: It does not - at least not when I ran this workflow. Maybe this was unintentional and it should be throwing an error.

Comment (tkkuehn) [reply scattr/workflow/Snakefile:72]: Took a look at the snakemake source and it seems like you're right, I'm not sure why I thought that

Comment (kaitj) [reply scattr/workflow/Snakefile:72]: I think there is an open PR or issue on the snakebids side for handling partial expansions that might be where this thought came from?

In any case, do we still want to make a change here or leave as is?

Comment (tkkuehn) [reply scattr/workflow/Snakefile:72]: If you're satisfied that it works as is, feel free to just leave it

Comment (kaitj) [reply scattr/workflow/Snakefile:72]: I will check the outputs again before resolving this to make sure the naming and stuff are still right.

**Review by tkkuehn — APPROVED** [review]
Needs a `snakefmt` but otherwise good to go

**Pull request merged**